src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -3353,6 +3353,42 @@ void mvnw(Register Rd, Register Rm,
   INSN(sve_whilelsw, 0b111, 0);
 #undef INSN
 
+private:
+
+  void encode_cvtf_T (SIMD_RegVariant& T_dst, SIMD_RegVariant& T_src) {
+    assert(T_src != B && T_dst != B &&
+           T_src != Q && T_dst != Q, "invalid register variant");
+    if (T_dst != D) {
+      assert(T_dst <= T_src, "invalid register variant");
+    } else {
+      assert(T_src != H, "invalid register variant");
+    }
+    if (T_dst == S && T_src == D) {
+      T_dst = D;
+      T_src = S;
+    } else if (T_dst == D && T_src == S) {
+      T_dst = D;
+      T_src = B;
+    }
+  }
+public:
+
+// SVE convert integer to floating-point (predicated)
+#define INSN(NAME, sign)                                                \
+  void NAME(FloatRegister Zd, SIMD_RegVariant T_dst, PRegister Pg,      \
+            FloatRegister Zn, SIMD_RegVariant T_src) {                  \
+    starti;                                                             \
+    f(0b01100101, 31, 24), f(0b010, 21, 19), f(0b101, 15, 13);          \
+    f(sign, 16);                                                        \
+    pgrf(Pg, 10), rf(Zn, 5), rf(Zd, 0);                                 \
+    encode_cvtf_T(T_dst, T_src);                                        \
+    f(T_dst, 23, 22), f(T_src, 18, 17);                                 \
```

Comment:
I would suggest to fill the bits in order, i.e. move f(T_dst, 23, 22) right after (... 31, 24), and then (21, 19), (18, 17)...

Reply:
`T_dst` and `T_src` will be changed after  `encode_cvtf_T`, so they cannot move there.

Reply:
I think it's fine to assign the values at the beginning. And maybe you can add more test cases to check different combinations.

Reply:
I have added some comments here to explain codes here.

Reply:
Agree with @nsjian that filling the bits in order which can make the code more clean. Thanks!

Reply:
OK. I will change this order.